edx_solutions_api_integration/courses/views.py

```diff
@@ -1832,6 +1832,141 @@ class CoursesMetrics(SecureAPIView):
+        cache_category = 'course_enrollments'
+        if org_id:
+            cache_category = '{}_{}'.format(cache_category, org_id)
+            if exclude_org_admins:
+                cache_category = '{}_exclude_admins'.format(cache_category)
+
+        enrollment_count = get_cached_data(cache_category, course_id)
+        if enrollment_count is not None:
+            return enrollment_count.get('enrollment_count')
+
+        course_key = get_course_key(course_id)
+        exclude_user_ids = get_aggregate_exclusion_user_ids(course_key)
+        users_enrolled_qs = CourseEnrollment.objects.users_enrolled_in(course_key).exclude(id__in=exclude_user_ids)
+
+        if org_id:
+            users_enrolled_qs = users_enrolled_qs.filter(organizations=org_id).distinct()
+            if exclude_org_admins:
+                non_company_users = get_non_actual_company_users('mcka_role_company_admin', org_id)
+                users_enrolled_qs.exclude(id__in=non_company_users)
+
+        enrollment_count = users_enrolled_qs.count()
+        cache_course_data(cache_category, course_id, {'enrollment_count': enrollment_count})
+
+        return enrollment_count
+
+    @staticmethod
+    def get_course_avg_grade(course_id, org_id=None):
+        """
+        Get average grade socre of a course
+        If org_id is passed then score is limited to that org's users
+        """
+        cache_category = 'grade'
+        if org_id:
+            cache_category = '{}_{}'.format(cache_category, org_id)
+
+        data = get_cached_data(cache_category, course_id)
+        if data is not None:
+            return data.get('course_avg')
+
+        course_key = get_course_key(course_id)
+        exclude_user_ids = get_aggregate_exclusion_user_ids(course_key)
+
+        avg_grade = StudentGradebook.course_grade_avg(
+            course_key,
+            exclude_users=exclude_user_ids,
+            org_ids=[org_id] if org_id else None
+        )
+        cache_course_data(cache_category, course_id, {'course_avg': avg_grade})
+
+        return avg_grade
+
+    @staticmethod
+    def get_course_avg_progress(course_id, org_id=None, percent_completion=True):
+        """
+        Get average progress score of a course
+        If org_id is passed then score is limited to that org's users
+        """
+        cache_category = 'progress'
+        if org_id:
+            cache_category = '{}_{}'.format(cache_category, org_id)
+
+        data = get_cached_data(cache_category, course_id)
+        if data is not None:
+            return data.get('course_avg')
+
+        course_key = get_course_key(course_id)
+        exclude_user_ids = get_aggregate_exclusion_user_ids(course_key)
+
+        data = _get_course_progress_metrics(
+            course_key,
+            exclude_users=exclude_user_ids,
+            org_ids=[org_id] if org_id else None,
+            percent_completion=percent_completion,
+        )
+        cache_course_data(cache_category, course_id, data)
```

Comment:
Since progress app is removed which is invalidating this cache on progress change, Progress will not change for 30 minutes is that fine?

Reply:
I think for aggregated scores, it should be fine it they delay by 30 minutes